lib/ws_servers/api/handlers/strategy/strategy_manager.js

```diff
@@ -173,116 +230,243 @@ class StrategyManager {
+    await liveStrategyExecutor.execute()
 
-    this._setActiveStatus(true)
+    const startedOn = Date.now()
+    this.strategy.set(strategyMapKey, {
+      strategy,
+      strategyOpts,
+      liveStrategyExecutor,
+      startedOn,
+      channels: this._getChannels(strategyOpts)
+    })
 
     const { name, symbol, tf } = strategy
     this._sendSuccess(
       `Started live strategy execution(${name}) for ${symbol}-${tf}`,
       ['startedLiveStrategyExecution', { name, symbol, tf }]
     )
+
+    this.pub(['strategy.live_execution_started', strategyMapKey, { ...strategyOpts, startedOn }])
+    this._sendLiveExecutionStatus()
   }
 
   /**
    * @public
-   * @returns {boolean}
+   * @returns {Boolean}
    */
-  isActive () {
-    return this.active
+  isActive (strategyMapKey) {
+    return this.strategy.has(strategyMapKey)
   }
 
   /**
    * @public
-   * @returns {object}
+   * @returns {Array[string]}
    */
-  getStrategyArgs () {
-    return this.strategyArgs
+  getActiveStrategies () {
+    const activeStrategies = [...this.strategy.entries()]
+    return activeStrategies.reduce((parsedStrat, strat) => {
+      const [key, value] = strat
+      const { strategyOpts, startedOn } = value
+      parsedStrat[key] = {
+        ...strategyOpts,
+        startedOn
+      }
+      return parsedStrat
+    }, {})
   }
 
   /**
    * @private
-   * @param {Object} [args] - strategy options
-   * @param {string} [args.id] - strategy id
-   * @param {string} [args.symbol] - symbol pair
-   * @param {string} [args.tf] - timeframe
-   * @param {boolean} [args.includeTrades] - option to include trades or not
-   * @param {number} [args.seedCandleCount] - number of candles to seed before strategy execution
    */
-  _setStrategyArgs (args = {}) {
-    this.strategyArgs = {
-      ...this.strategyArgs,
-      ...args
+  _clearStrategy (strategyMapKey) {
+    const strategyExists = this.strategy.has(strategyMapKey)
+    if (!strategyExists) {
+      return
     }
+    this.strategy.delete(strategyMapKey)
+  }
+
+  _unsubscribe (unsubscriptionChannel, filter) {
+    this.ws2Manager.withDataSocket(({ channel, ...restArgs }) => {
+      const fv = _pick(restArgs, Object.keys(filter))
+      return unsubscriptionChannel === channel && _isEqual(fv, filter)
+    }, (socket) => {
+      const chanId = findChannelId(socket, (data) => {
+        if (data.channel !== unsubscriptionChannel) {
+          return false
+        }
+
+        const fv = _pick(data, Object.keys(filter))
+        return _isEqual(filter, fv)
+      })
+
+      if (!chanId) {
+        return this.d('error unsubscribing: unknown %s channel [%o]', unsubscriptionChannel, filter)
+      }
+
+      const nextSocket = unsubscribe(socket, chanId)
+
+      return nextSocket
+    })
   }
 
   /**
    * @private
-   * @param {boolean} status
    */
-  _setActiveStatus (status) {
-    this.active = status
+  _unsubscribeChannels (channels, strategyMapKey) {
```

Comment:
both `_unsubscribe` and `_unsubscribeChannels` have a lot of conditional branches and corner cases, I would recommend increasing the test coverage for that